ods_ci/run_robot_test.sh

```diff
@@ -319,12 +319,29 @@ if command -v yq &> /dev/null
 fi
 
 if [[ ${SKIP_INSTALL} -eq 0 ]]; then
-  # look for pre-created poetry .venv
-  virtenv="${HOME}/.local/ods-ci/.venv"
-  if [[ -d "${virtenv}" ]]; then
+  # Look for pre-created poetry .venv
+  # NOTE: handle that checkout might happen through `git checkout -f somehash` and branch names may include `/` char
+  refname="$(git branch --remote --no-abbrev --format="%(refname:lstrip=3)" --contains "HEAD" | head -1)"
+  venvdir="${refname%,*}"
+  echo "Git revision refname='${refname}', venvdir='${venvdir}'."
+  for venvdir in "${venvdir}" "master"; do
+    virtenv="${HOME}/.local/ods-ci/${venvdir}/.venv"
+    echo "Checking whether '${virtenv}' exists."
+    if [[ -d "${virtenv}" ]]; then
+      break
+    fi
+  done
+  if [[ -d "${basepath}/../.venv" ]]; then
+    echo "We ran already before and settled in venv we linked into '${basepath}/../.venv'."
+  elif [[ -d "${virtenv}" ]]; then
     echo "Using a pre-created virtual environment in '${virtenv}' for poetry to save time."
     poetry config --local virtualenvs.in-project true
+    # Workaround for Poetry urllib3 connection error
+    poetry config --local installer.parallel false
```

Comment:
I'm wondering whether we still want this workaround, but I suppose that it doesn't matter that much if the venvs will be in place in the most of the time this is executed.

Reply:
I think we do, the issue in poetry was not resolved and because the .venvs are in place, installing serially will not slow us much.

Reply:
Yes, I'm aware. My point for removing was because we have the .venvs ready for different branches so it's less probable to hit. But yeah, let's keep it there for the cache-miss situations.